account_asset_management_no_depreciation/models/account_asset.py

```diff
@@ -0,0 +1,55 @@
+# Copyright 2021 Ecosoft Co., Ltd. (http://ecosoft.co.th)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+READONLY_STATES = {
+    "open": [("readonly", True)],
+    "close": [("readonly", True)],
+    "removed": [("readonly", True)],
+}
+
+
+class AccountAsset(models.Model):
+    _inherit = "account.asset"
+
+    no_depreciation = fields.Boolean(
+        string="Non-Depreciation",
+        compute="_compute_no_depreciation",
+        readonly=False,
+        store=True,
+        states=READONLY_STATES,
+        help="Check this if you do not want to compute depreciation.",
+    )
+
+    @api.depends("profile_id")
+    def _compute_no_depreciation(self):
+        for asset in self:
+            asset.no_depreciation = asset.profile_id.no_depreciation
+
+    @api.onchange("no_depreciation")
+    def _onchange_no_depreciation(self):
+        if not self.no_depreciation:
+            self.method_number = 0
+            self.salvage_value = 0.0
+
+    @api.constrains("no_depreciation", "method_number")
+    def _check_method(self):
+        if any(a.no_depreciation and a.method_number != 0 for a in self):
+            raise UserError(
+                _("Non-Depreciation is only supported for Number of Years = 0.")
+            )
```

Comment:
I think if we require method_number == 0, we don't need following.
```suggestion
```

Reply:
Rather than, trying to ensure that method_number = 0, I think it is better to ensure that, the compute_depreciation_board is not called. Less lines of code to maintain too.

WDYT?

```
    def compute_depreciation_board(self):
        assets_no_depre = self.filtered("no_depreciation")
        return super(AccountAsset, self-asset_no_depre).compute_depreciation_board()
```